xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -463,3 +483,105 @@ func circuitBreakersFromCluster(cluster *v3clusterpb.Cluster) *uint32 {
 	}
 	return nil
 }
+
+// outlierConfigFromCluster extracts the relevant outlier detection configuration from
+// the received cluster resource. Returns nil if no OutlierDetection.
+func outlierConfigFromCluster(cluster *v3clusterpb.Cluster) (*OutlierDetection, error) {
+	// Corresponds to this line from A50: "If the outlier_detection field is not
+	// set in the Cluster message, a "no-op" outlier_detection config will be
+	// generated"
+	od := cluster.GetOutlierDetection()
+	if od == nil {
+		return nil, nil
+	}
+	// "The google.protobuf.Duration fields interval, base_ejection_time, and
+	// max_ejection_time must obey the restrictions in the
+	// google.protobuf.Duration documentation and they must have non-negative
+	// values." - A50
+	// TODO: I can find non negative examples for validation, but not "must obey the restrictions in the documentation"
+	var iVal, betVal, metVal = defaultInterval, defaultBaseEjectionTime, defaultMaxEjectionTime
```

Comment:
The `Val` suffix in these variable names does not add much value. I would prefer the following instead:

```golang
interval, baseEjectionTime, maxEjectionTime := defaultInterval, defaultBaseEjectionTime, defaultMaxEjectionTime
```

These are longer than your names. But given that this function is a reasonably long one, longer names are fine.

From the style guide:
```
The general rule of thumb is that the length of a name should be proportional to the size of its scope and inversely 
proportional to the number of times that it is used within that scope. A variable created at file scope may require 
multiple words, whereas a variable scoped to a single inner block may be a single word or even just a character or 
two, to keep the code clear and avoid extraneous information.
```